header-only/awkward/LayoutBuilder.h

```diff
@@ -2247,12 +2247,16 @@ namespace awkward {
       clear() noexcept {
         mask_.clear();
         content_.clear();
+        current_byte_ = 0;
+        current_byte_ref_ = mask_.append_and_get_ref(current_byte_);
+        current_index_ = 0;
       }
 
       /// @brief Current length of the `mask` buffer.
       size_t
       length() const noexcept {
-        return (mask_.length() - 1) * 8 + current_index_;
+        return mask_.length() > 0 ?
+          (mask_.length() - 1) * 8 + current_index_ : current_index_;
```

Comment:
Good catch!

I think `length()` is signed, so `length() - 1` doesn't roll over to `MAX_INT` (which would be even worse), but it still return a nonsensical (negative) result if `length()` is zero.

Oh—it's not! (In the ArrayBuilder code, I avoided the platform-dependent `size_t` and used `int64_t` instead. An argument could be made for using all `size_t` or all `int64_t`. The worst situation is using some here, some there. Compilers on 64-bit platforms don't catch it and may or may not warn about the signed-unsigned comparisons.)

So this really is fixing a worst case scenario: `length() - 1` is a crazy huge number when `length()` is zero.

Reply:
> Were the changes in GrowableBuffer.h and LayoutBuilder.h needed because the tests hadn't been running and the code became out-of-date?
> 
No, it's because we did not test `clear()` and `length()` methods consistently. Now the test has lots of `asserts` :-)

> Anyway, it's much better now that they are. Go ahead and merge this PR!

Thanks!